pkg/cli/iam_cleanup.go

```diff
@@ -0,0 +1,178 @@
+	testHandler iamCleanupHandler
+}
+
+func (c *IAMCleanupCommand) Desc() string {
+	return `Cleanup of the IAM request YAML file in the given path, which ` +
+		`removes bindings in the request and expired AOD bindings from the IAM policy`
+}
+
+func (c *IAMCleanupCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+Cleanup of the IAM request YAML file in the given path:
+
+      {{ COMMAND }} -path "/path/to/file.yaml"
+
+Cleanup of the IAM request YAML file and output applied IAM changes:
+
+      {{ COMMAND }} -path "/path/to/file.yaml" -verbose
+`
+}
+
+func (c *IAMCleanupCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+
+	// Command options
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "path",
+		Target:  &c.flagPath,
+		Example: "/path/to/file.yaml",
+		Predict: predict.Files("*"),
+		Usage:   `The path of IAM request file, in YAML format.`,
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "verbose",
+		Target:  &c.flagVerbose,
+		Default: false,
+		Usage:   `Turn on verbose mode to print updated IAM policies. Note that it may contain sensitive information`,
+	})
+
+	return set
+}
+
+func (c *IAMCleanupCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %q", args)
+	}
+
+	if c.flagPath == "" {
+		return fmt.Errorf("path is required")
+	}
+
+	return c.cleanupIAM(ctx)
+}
+
+func (c *IAMCleanupCommand) cleanupIAM(ctx context.Context) error {
+	// Read request from file path.
+	var req v1alpha1.IAMRequest
+	if err := requestutil.ReadRequestFromPath(c.flagPath, &req); err != nil {
+		return fmt.Errorf("failed to read %T: %w", &req, err)
+	}
+
+	if err := v1alpha1.ValidateIAMRequest(&req); err != nil {
+		return fmt.Errorf("failed to validate %T: %w", &req, err)
+	}
+
+	var h iamCleanupHandler
+	if c.testHandler != nil {
+		// Use testHandler if it is for testing.
+		h = c.testHandler
+	} else {
+		// Create resource manager clients.
+		organizationsClient, err := resourcemanager.NewOrganizationsClient(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to create organizations client: %w", err)
+		}
+		defer organizationsClient.Close()
+
+		foldersClient, err := resourcemanager.NewFoldersClient(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to create folders client: %w", err)
+		}
+		defer foldersClient.Close()
+
+		projectsClient, err := resourcemanager.NewProjectsClient(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to create projects client: %w", err)
+		}
+		defer projectsClient.Close()
+
+		// Create IAMHandler with the clients.
+		h, err = handler.NewIAMHandler(
+			ctx,
+			organizationsClient,
+			foldersClient,
+			projectsClient,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create IAM handler: %w", err)
+		}
+	}
```

Comment:
It seems you can refactor this out to share with iam_handle.go?

Reply:
Yes, read request from file path + request validation + NewIAMHandler are the same, testHandler uses a different interface. I think there is no need to refactor `read request from file path + request validation`, we can refactor `NewIAMHandler `.